Added initial account banned modal

TheHollidayInn opened the pull request:
Fixes #8636 
Fixes #9841
Fixes #8672

TheHollidayInn committed:
Added initial account banned modal

paglias commented on website/client/app.vue:
```diff
@@ -370,6 +377,19 @@ export default {
   },
   methods: {
+    checkForBannedUser (error) {
+      const AUTH_SETTINGS = localStorage.getItem('habit-mobile-settings');
```
looks good, just one thing: why are we parsing the auth settings again if these are stored into `store.state.credentials`? Because they cannot login at all when banned so the credentials are never stored?

TheHollidayInn replied:
Hmmm. I figured a user would be banned while logged in but I'm sure both cases would happen. For testing locally, I banned my user while he was logged in.

Alys commented:
Users are most often banned while they are logged in.

On Wed, 24 Jan 2018, 08:37 Keith Holliday <notifications@github.com> wrote:

> *@TheHollidayInn* commented on this pull request.
> ------------------------------
>
> In website/client/app.vue
> <https://github.com/HabitRPG/habitica/pull/9868#discussion_r163399686>:
>
> > @@ -370,6 +377,19 @@ export default {
>      if (loadingScreen) document.body.removeChild(loadingScreen);
>    },
>    methods: {
> +    checkForBannedUser (error) {
> +      const AUTH_SETTINGS = localStorage.getItem('habit-mobile-settings');
>
> Hmmm. I figured a user would be banned while logged in but I'm sure both
> cases would happen. For testing locally, I banned my user while he was
> logged in.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitica/pull/9868#discussion_r163399686>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbTAc0w1ZjI3ItJ8RsUFpSqSzK-K041ks5tNl9BgaJpZM4RqOYB>
> .
>

TheHollidayInn commented:
Sweet. So is this PR good then besides the need to remove the profile name?

paglias commented:
LGTM 👍

Alys commented on website/common/locales/en/front.json:
```diff
@@ -285,6 +285,8 @@
   "invalidCredentials": "There is no account that uses those credentials.",
   "accountSuspended": "Account has been suspended, please contact <%= communityManagerEmail %> with your User ID \"<%= userId %>\" for assistance.",
+  "accountSuspendedTitle": "Account has been suspended",
+  "accountSuspendedBody": "This account, Profile Name <%= profileName %>, User ID \"<%= userId %>\", has been blocked for breaking the [Community Guidelines](https://habitica.com/static/community-guidelines) or [Terms of Service](https://habitica.com/static/terms). For details or to ask to be unblocked, please email Lemoness at <%= communityManagerEmail %> or ask your parent or guardian to email  ...
```
Suggested new text, which removes the profile name and also Lemoness's name so the message still works at times when she's on leave:

```
"This account, User ID \"<%= userId %>\", has been blocked for breaking the [Community Guidelines](https://habitica.com/static/community-guidelines) or [Terms of Service](https://habitica.com/static/terms). For details or to ask to be unblocked, please email our Community Manager at <%= communityManagerEmail %> or ask your parent or guardian to email them. Please copy your User ID into the email and include your Profile Name.", 
```

Also can the new text be copied into the `accountSuspended` key and have the code use that instead of `accountSuspendedBody`? That means the website and API/mobile apps will be using the same message and it also keeps the history of the message in one place in git's log, which will be useful if we ever need to look back through previous versions of the message (complaints from blocked users might occasionally make that useful).

Alys reviewed (changes requested):
This is looking great for the case where the user is already logged in! I love the new modal - very clear and easy for the user to copy out their User ID.

However if the user is already logged out and tries to log in after they've been blocked, no message comes up - the "Login" button lets you click it and causes a sync but there's no feedback about why you can't log in.

TheHollidayInn committed:
Merged in develop

TheHollidayInn committed:
Fixed check for non logged in user

TheHollidayInn commented:
Okay, updated the copy and added a check for not logged in. 

We don't have the UUID when the user is not logged in, so I can't reliably check for the error message comparison. There is a 'growl' message that shows the normal response error although it isn't the best.

TheHollidayInn committed:
Merge remote-tracking branch 'upstream/develop' into account-banned-modal

Alys commented:
These parts are working as required:
- If you become blocked while you are using Habitica, you see the modal.
- If you reload, you see the modal again.

There's still the problem where if you try to log in, you don't see a modal, you only see the growl notification, but I see your point above about not being able to get the User ID. It would be okay if that modal didn't have the User ID in it. For example, it could say:
"This account has been blocked for breaking the Community Guidelines or Terms of Service. For details or to ask to be unblocked, please email our staff at admin@habitica.com or ask your parent or guardian to email them. Please include your Profile Name."

However if you prefer, this PR could be merged as it is and the login modal could be added later. This PR is definitely an improvement on the current situation!

There's also a problem where if you're logged in at the time you are blocked, you can't get to the clear-browser-data page, which means it's difficult for another user to use Habitica on the same PC, but that problem also exists in `develop` so it could be fixed later too. We might want to discuss putting a logout button on the modal, although that might give the user too much of a hint about creating a new account to bypass the ban.

TheHollidayInn commented:
If it is fine with you, I think we can start with this and iterate. This is a pretty good improvement.

 And I agree about the logout. I'd rather them email us rather than pushing them to create a new account.

Alys commented:
I'm fine with it being included in our next batch of PRs to be merged! I've marked it ready for review in case anyone wants to do a code review again.

TheHollidayInn committed:
Merge branch 'develop' into account-banned-modal
